deeppavlov/models/elmo/elmo_model.py

```diff
@@ -0,0 +1,709 @@
+                        pass
+                    else:
+                        # add a skip connection
+                        lstm_cell = tf.nn.rnn_cell.ResidualWrapper(lstm_cell)
+
+                # collect the input state, run the dynamic rnn, collect
+                # the output
+                # the LSTMs are stateful.  To support multiple batch sizes,
+                # we'll allocate size for states up to max_batch_size,
+                # then use the first batch_size entries for each batch
+                init_states = [
+                    tf.Variable(
+                        tf.zeros([self._max_batch_size, dim]),
+                        trainable=False
+                    )
+                    for dim in lstm_cell.state_size
+                ]
+                batch_init_states = [
+                    state[:batch_size, :] for state in init_states
+                ]
+
+                if direction == 'forward':
+                    i_direction = 0
+                else:
+                    i_direction = 1
+                variable_scope_name = 'RNN_{0}/RNN/MultiRNNCell/Cell{1}'.format(
+                    i_direction, i)
+                with tf.variable_scope(variable_scope_name):
+                    layer_output, final_state = tf.nn.dynamic_rnn(
+                        lstm_cell,
+                        layer_input,
+                        sequence_length=sequence_lengths,
+                        initial_state=tf.nn.rnn_cell.LSTMStateTuple(
+                            *batch_init_states),
+                    )
+
+                self.lstm_state_sizes[direction].append(lstm_cell.state_size)
+                self.lstm_init_states[direction].append(init_states)
+                self.lstm_final_states[direction].append(final_state)
+                if direction == 'forward':
+                    self.lstm_outputs[direction].append(layer_output)
+                else:
+                    self.lstm_outputs[direction].append(
+                        tf.reverse_sequence(
+                            layer_output,
+                            sequence_lengths,
+                            seq_axis=1,
+                            batch_axis=0
+                        )
+                    )
+
+                with tf.control_dependencies([layer_output]):
+                    # update the initial states
+                    for i in range(2):
+                        new_state = tf.concat(
+                            [final_state[i][:batch_size, :],
+                             init_states[i][batch_size:, :]], axis=0)
+                        state_update_op = tf.assign(init_states[i], new_state)
+                        update_ops.append(state_update_op)
+    
+                layer_input = layer_output
+
+        self.mask = mask
+        self.sequence_lengths = sequence_lengths
+        self.update_state_op = tf.group(*update_ops)
+
+
+def weight_layers(name, bilm_ops, l2_coef=None,
```

Comment:
deeppavlov/models/elmo/weight_layers.py?

Reply:
moved to deeppavlov/models/elmo/elmo_model.py